expat/tests/common.c

```diff
@@ -190,6 +191,10 @@ _xml_failure(XML_Parser parser, const char *file, int line) {
 enum XML_Status
 _XML_Parse_SINGLE_BYTES(XML_Parser parser, const char *s, int len,
                         int isFinal) {
+  // This ensures that tests have to run pathological parse cases
+  // (e.g. when `s` is NULL) against plain XML_Parse rather than
+  // chunking _XML_Parse_SINGLE_BYTES.
+  assert((parser != NULL) && (s != NULL) && (len >= 0));
```

Comment:
I preferred where we were going with the other approach, but this is also fine (as long as there are no tests that need to do `if NULL Parse else Parse_SINGLE`, but that doesn't seem to be the case right now).